### ISSUE TITLE
Bug: links does not show the page and redirect to the react props names

### ISSUE THREAD

**Issue by a0m0rajab** [open-sauced/app#2185]
### Describe the bug

When I tried to use the link from this issue: https://github.com/open-sauced/app/issues/2177
I was redirected to https://app.opensauced.pizza/[pageId]/[toolName]/filter/[...selectedFilter]?range=30 
Instead of:  https://app.opensauced.pizza/ai/dashboard/filter/top-100-repos?range=30

### Steps to reproduce

1. go to [app.opensauced.pizza/ai/dashboard/filter/top-100-repos?range=30](https://app.opensauced.pizza/ai/dashboard/filter/top-100-repos?range=30)
2. Tada 🎉 

### Browsers

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

**Comment by nickytonline** [comment]
I have a feeling the info is not being read from the URL, e.g. language.

**Comment by a0m0rajab** [comment]
When I tested this, the info was read rendered and then redirected to this weird names, I did not dig further into it to be honest.

Sent from mobile device


________________________________

**Comment by nickytonline** [comment]
I looked into this a bit and it appears to be the `?range=30` that causes it to behave the way the bug is described. However, if you navigate to https://app.opensauced.pizza/javascript/dashboard/filter/recent without the querystring `?range=30`, it's all good.

This is probably related to changes in #2047.

**Comment by a0m0rajab** [comment]
Could I work on this? I would like to debug this.  

just trying if .take works.

**Comment by nickytonline** [comment]
I've assigned it to you @a0m0rajab . @OgDev-01 worked on

**Comment by a0m0rajab** [comment]
Dev notes: 
When I opened [app.opensauced.pizza/javascript/dashboard/filter/recent](https://app.opensauced.pizza/javascript/dashboard/filter/recent) it just gave the wrong names then get back to the normal one. I think it's a useEffect issue.

**Comment by JabSYsEmb** [comment]
I just found the function call that causing in the redirecting after the page being rendered, I share it here for further investigation
![B3sQK0z](https://github.com/open-sauced/app/assets/32222578/af21047a-9e26-463c-962c-ec96d94bc3c5)

**Comment by a0m0rajab** [comment]
Thanks @JabSYsEmb for checking this, if you could find the related code in the dev it would be great, I could find that. 

As for my investigation today: 
- the [app.opensauced.pizza/javascript/dashboard/filter/recent](https://app.opensauced.pizza/javascript/dashboard/filter/recent) link is working then it shows the broken URL then it works again. 
- the URL with range=30 does work then get broken and show the broken page. 


The issue is happening in the next function, if we remove the merge logic it will work well. 
```js
function setQueryParams(params: Record<string, string>, paramsToRemove: Array<string> = [], pathname?: string) {
  const presentQueryParams = deleteKeys(Router.query, paramsToRemove);
  debugger;
  // Merge final query and set URL
  Router.push({ 
    pathname: pathname || Router.pathname,
     query: {
       ...presentQueryParams,
       ...params,
     },
   });
}
```

**Comment by a0m0rajab** [comment]
I found the issue, I think it was due the prerendering since the next provides and empty query object. 
ref: https://nextjs.org/docs/pages/api-reference/functions/use-router

I will open a PR in few minutes.


Another few issues I noticed while working on this, I will check them later and open an issue for them: 
- the tool selector is showing an empty page if no tool were selected: related code `case "Dashboard":`
- There is a warning related to style when you open the app in the debugger: 
```bash
Warning: Updating a style property during rerender (overflowY) when a conflicting property is set (overflow) can lead to styling bugs. To avoid this, don't mix shorthand and non-shorthand properties for the same value; instead, replace the shorthand with separate values.
```
- the name of the range selector should be limit to set the information 
- Range is not setting in the page it's just showing 10 per page always. 
- when you click on showing 10 per page it will break the layout a bit 
- when limit/range is selected this does not appear on the button in the UI
